### PR TITLE
[integration-test] Introduce an environment package

### PR DESCRIPTION
**Pull request by vdemeester**
This would allow to split responsabilities and require less global variables. This way, we would have one and only one global variable `testEnv` that can have behavior (like `ensureFrozenImages`, …).

- This removes `docker_test_vars.go` and `docker_utils.go` `init()` functions to keep things sane.
- This does not remove all the global variables for now to keep the diff light.
- There is still a few improvements to do (on `protected` images)

/cc @dnephin @thaJeztah @icecrime @cpuguy83 @tiborvass 

🐸

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

### CODE REVIEW

**Review by dnephin — APPROVED** [review]
LGTM

File: integration-cli/check_test.go [hunk 1]
```diff
@@ -4,27 +4,136 @@ import (
...
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	assignGlobalVariablesFromTestEnv(testEnv)
+}
+
+// FIXME(vdemeester) remove this and use environment
+func assignGlobalVariablesFromTestEnv(testEnv *environment.Execution) {
+	fmt.Printf("%+v\n", testEnv)
```
Comment (tiborvass) [inline integration-cli/check_test.go]: is this debug intentional?

**Comment by tiborvass** [comment]
I'd like to wait at least for rc5 before merging this.

File: integration-cli/check_test.go (hunk 1 above)
Comment (vdemeester) [inline integration-cli/check_test.go]: ah right yeah, I should remove that one.. I was still wondering how to handle protected images but yeah this printf should be removed :sweat_smile:

**Review by LK4D4 — APPROVED** [review]
LGTM

**Comment by thaJeztah** [comment]
all green!

**Pull request merged**